### PR TITLE
add: remanga.org

### PR DESCRIPTION
**Pull request by knst0**
Support for https://remanga.org

### CODE REVIEW

File: src/web/mjs/connectors/ReManga.mjs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        const request = new Request(new URL('/api/titles/' + slug, this.api), this.requestOptions);
+        const { content } = await this.fetchJSON(request, 3);
+
+        return new Manga(this, content.dir, content.en_name || content.another_name);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+
+        // Of course, I can get the all manga, but it's over 20 000
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs]: Remove comment

File: src/web/mjs/connectors/ReManga.mjs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        for (let page = 0; page <= 100; page++) {
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs:28]: Get full list or throw an expressive error, e.g.
https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/NHentai.mjs#L25

File: src/web/mjs/connectors/ReManga.mjs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        }
+
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const requestUrl = new URL('/api/search/catalog/', this.api);
+
+        requestUrl.searchParams.set('ordering', '-rating');
+        requestUrl.searchParams.set('count', '30');
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs:40]: May try to increase `count` to get more mangas per page

File: src/web/mjs/connectors/ReManga.mjs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL('/api/titles/' + manga.id, this.api), this.requestOptions);
+        const { content } = await this.fetchJSON(request, 3);
+
+        const branch = content.branches[0];
+        const pagesCount = content.count_chapters > 60 ? Math.trunc(content.count_chapters / 60) : 1;
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs]: Is this behavior correct?
`content.count_chapters = 50` and `content.count_chapters = 100` will both result in `1`

File: src/web/mjs/connectors/ReManga.mjs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            chapters.push(...tmp);
+        }
+        return chapters;
+    }
+
+    async _getChaptersFromPage(branchId, page) {
+        const requestUrl = new URL('/api/titles/chapters/', this.api);
+
+        requestUrl.searchParams.set('branch_id', branchId);
+        requestUrl.searchParams.set('count', '60');
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs]: May try to increase `count` to get more chapters per page

File: src/web/mjs/connectors/ReManga.mjs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        requestUrl.searchParams.set('ordering', '-index');
+        requestUrl.searchParams.set('user_data', '1');
+        requestUrl.searchParams.set('page', page || '1');
+
+        const request = new Request(requestUrl, this.requestOptions);
+        const data = await this.fetchJSON(request);
+
+        const filterPaidChapters = data.content.filter(item => !(item.is_paid && !item.is_bought));
```
Comment (27136773) [inline src/web/mjs/connectors/ReManga.mjs]: May be simplified by removing brackets `!item.is_paid || item.is_bought`

**Review by 27136773 — COMMENTED** [review]
Revise/verify at least the calculation of chapter page count

File: src/web/mjs/connectors/ReManga.mjs (hunk 3 above)
Comment (knst0) [reply src/web/mjs/connectors/ReManga.mjs:40]: `30` is maximum

File: src/web/mjs/connectors/ReManga.mjs (hunk 5 above)
Comment (knst0) [reply src/web/mjs/connectors/ReManga.mjs]: Changed to `80`, is maximum

**Review by 27136773 — APPROVED** [review]
Ready for merge

**Pull request merged**